components/match2/commons/match_group_display.lua

```diff
@@ -38,18 +38,19 @@ function MatchGroupDisplay.MatchlistBySpec(args, matchBuilder)
 		}, matches)
 	end
 
-	return table.concat(Array.extend(
-		tostring(matchlistNode),
-		Array.map(warnings, MatchGroupDisplay.WarningBox)
-	), '')
+	local parts = Array.extend(
+		Array.map(optionsWarnings, MatchGroupDisplay.WarningBox),
+		{matchlistNode}
+	)
+	return table.concat(Array.map(parts, tostring))
```

Comment:
Is the `tostring` correct here?

Reply:
yes, but only because Array.map doesn't pass the array index as the 2nd argument

I'll change it to be more explicit

Reply:
nvm, what's there is fine. I forgot tostring only takes one argument